bosond.cpp

```diff
@@ -77,13 +65,69 @@ void processArgs(int argc, char **argv) {
     }
 }
 
+// Initalise the command control interface (CCI) using SDK.
+void initCCI() {
+    libusb_device_handle* dev_h;
+
+    if (libusb_init(NULL) < 0) {
+        perror("Error: Failed libusb init");
+        exit(1);
+    }
+
+    if ((dev_h = libusb_open_device_with_vid_pid(NULL, 0x09cb, 0x4007)) == NULL) {
+        perror("couldn't open USB device");
+        exit(1);
+    }
+
+    Initialize(dev_h);
+}
+
+int logCameraInfo() {
+    uint32_t major, minor, patch;
+    if (int result = bosonGetSoftwareRev(&major, &minor, &patch)) {
+        std::cout << "failed to get software rev: " << result << '\n';
+        return -1;
+    }
+
+    std::cout << "Boson firmware version: " << major << '.' << minor << '.' << patch << '\n';
+    uint32_t camera_sn;
+    if (int result = bosonGetCameraSN(&camera_sn)) {
+        std::cout << "failed to get serial number: " << result << '\n';
+        return -1;
+    }
+    std::cout << "Boson serial: " << camera_sn << '\n';
+
+    return 0;
+}
+
+int sendAll(int sock, const char *data, size_t len) {
+    int left = len;
+    int n;
+
+    while (left > 0) {
+        n = send(sock, data, left, 0);
+        if (n < 0) {
+            return n;
+        }
+        left -= n;
+        data += n;
+    }
+    return 0;
+}
+
+
 
 int main(int argc, char** argv) {
     int fd;
     struct v4l2_capability cap;
 
     processArgs(argc, argv);
 
+    initCCI();
+    if (logCameraInfo()) {
```

Comment:
as above will this work with -1?

Reply:
Yep. Any integer value except 0 evaluates to true.